### PR TITLE
Cleanup inference to enable clean DDC compile

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: lib/src/json_serializable_generator.dart [hunk 1]
```diff
@@ -76,7 +76,7 @@ class JsonSerializableGenerator
     // Explicitly using `LinkedHashMap` – we want these ordered.
     var fields = new LinkedHashMap<String, FieldElement>.fromIterable(
         fieldsList,
-        key: (f) => f.name);
+        key: (FieldElement f) => f.name);
```
Comment (natebosch) [inline lib/src/json_serializable_generator.dart]: [aside] I'm a little surprised this one is necessary - is fieldsList not well typed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
-## 0.1.1
+## 0.2.0
 
-* Fail generation when undefined types are encountered.
-  Throw a helpful error.
+* **BREAKING** Fail generation for unsupported types.
```
Comment (natebosch) [inline CHANGELOG.md]: elaborate on what is unsupported

File: lib/src/json_serializable_generator.dart (hunk 1 above)
Comment (kevmoo) [inline lib/src/json_serializable_generator.dart]: Agreed: https://github.com/dart-lang/sdk/issues/26392

File: CHANGELOG.md (hunk 2 above)
Comment (kevmoo) [inline CHANGELOG.md]: done

**Pull request merged**